## Урок 1
### Случайные события. Условная вероятность. Формула Байеса.  Независимые испытания

***1. Из колоды в 52 карты извлекаются случайным образом 4 карты.***

***a) Найти вероятность того, что все карты – крести.***

***б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.***

а) **А** - все 4 карты имеют масть "крести".

Вероятность наступления события:

$$P(A) = \frac{m}{n}$$

Количество сочетаний определяется формулой:

$$C_n^k = \frac{n!}{k!(n - k)!}$$

Функция для вычисления количества сочетаний:

In [24]:
from math import factorial

In [25]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

В колоде из 52 карт - 13 карт масти "крести".

Количество благоприятствующих исходов (сочетания по 4 карты масти "крести"):

$$m = C_{13}^4$$

In [26]:
m = combinations(13, 4)
m

715

Общее количество исходов (сочетания по 4 карты из всей колоды):

$$n = C_{52}^4$$

In [27]:
n = combinations(52, 4)
n

270725

Вероятность того, что все 4 карты окажутся масти "крести":

$$P(A) = \frac{m}{n} = \frac{715}{270725} = 0,002641$$

б) **B** - среди четырех карт есть хотя бы один туз.

Число сочетаний, если среди четырех карт есть 1 туз: $$C_1 = C_4^1 * C_{48}^3 = 4 * 17296 = 69184$$

In [28]:
c1 = combinations(4, 1) * combinations(48, 3)
c1

69184

Число сочетаний, если среди четырех карт есть 2 туза: $$C_2 = C_4^2 * C_{48}^2 = 6 * 1128 = 6768$$

In [29]:
c2 = combinations(4, 2) * combinations(48, 2)
c2

6768

Число сочетаний, если среди четырех карт есть 3 туза: $$C_3 = C_4^3 * C_{48}^1 = 4 * 48 = 192$$

In [30]:
c3 = combinations(4, 3) * combinations(48, 1)
c3

192

Число сочетаний, если среди четырех карт есть 3 туза: $$C_4 = C_4^4 * C_{48}^0 = 1 * 1 = 1$$

In [31]:
c4 = combinations(4, 4) * combinations(48, 0)
c4

1

Количество благоприятствующих исходов (число сочетаний, если среди четырех карт есть хотя бы один туз): $$m = C_1 + C_2 + C_3 + C_4 = 69184 + 6768 + 192 + 1 = 76145$$

In [32]:
m = c1 + c2 + c3 + c4
m

76145

Общее количество возможных исходов (сочетания по 4 карты из всей колоды):

$$n = C_{52}^4$$

In [33]:
n = combinations(52, 4)
n

270725

Вероятность того, что среди 4 карт есть хотя бы 1 туз:

$$P(B) = \frac{m}{n} = \frac{76145}{270725} = 0,0028$$

***2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?***

**А** - угадать 3 цифры кода из 3 с первой попытки

Вероятность наступления события:

$$P(A) = \frac{m}{n}$$

Количество благоприятствующих исходов (нажать 3 цифры из 3 правильных):

$$m = C_{3}^3 = 1$$

In [34]:
m = combinations(3, 3)
m

1

Общее количество возможных исходов:

$$n = C_{10}^3 = 120$$

In [35]:
m = combinations(10, 3)
m

120

Вероятность открыть дверь с первой попытки:

$$P(A) = \frac{m}{n} = \frac{1}{120}$$

***3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?***

**А** - все извлеченные детали окрашены.

Вероятность наступления события:

$$P(A) = \frac{m}{n}$$

Количество благоприятствующих исходов (сочетание по 3 детали из 9 окрашенных):

$$m = C_{9}^3 = 84$$

In [36]:
m = combinations(9, 3)
m

84

Общее количество возможных исходов (сочетание по 3 детали из 15):

$$n = C_{15}^3 = 455$$

In [37]:
m = combinations(15, 3)
m

455

Вероятность, что все извлеченные детали будут окрашены:

$$P(A) = \frac{m}{n} = \frac{84}{455} = 0,1846$$

***4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?***

**А** - оба приобретенных билета окажутся выигрышными.

Вероятность наступления события:

$$P(A) = \frac{m}{n}$$

Количество благоприятствующих исходов (2 приобретенных билета выигрышные):

$$m = C_{2}^2 = 1$$

In [38]:
m = combinations(2, 2)
m

1

Общее количество возможных исходов (число комбинаций по 2 приобретенных билета из 100):

$$n = C_{100}^2 = 4950$$

In [39]:
m = combinations(100, 2)
m

4950

Вероятность, что оба приобретенных билета окажутся выигрышными:

$$P(A) = \frac{m}{n} = \frac{1}{4950} = 2,02 * 10^{-4}$$